pyclaragenomics/claragenomics/bindings/cudapoa.pyx

```diff
@@ -65,11 +65,11 @@ cdef class CudaPoaBatch:
 
     def __cinit__(
             self,
-            max_sequences_per_poa,
-            gpu_mem,
+            max_mem,
+            output_mask,
+            batch_size,
```

Comment:
instead of exposing batch size as a struct, let's add the member variables as optional parameters to the API (similar to how gap_score and all the setup). so something like 

```
def __cinit__ (
self,
max_sequences_per_poa,
max_seq_size,
max_concensus_size,
max_mem,
output_mask,
....
max_nodes_per_poa,
)
```

and then send the `nodes_per_group` parameter to both banded and non banded paramter. I think in the C++ API as well we can keep just one and use the banded parameter to set the appropriate one internally - wdyt @r-mafi ?

Reply:
Don't you think using BatchSize structure will make the binding immune to changes in C++ size (at-lease reduced ripple-effect)?

Reply:
what I'm missing here is in C++, we have a constructor that assigns proper values to most of `BatchSize` members based on a minimal set of inputs (only `max_seq_sz` and `max_seq_per_poa`). How is that done here? Should user adjust `max_consensus_size` according to arbitrary `max_seq_size`? Are you suggesting to expose all `BatchSize` struct members or only a minimal set?

Reply:
I realized I had written that comment incorrectly. updated it.

@r-mafi - exposing only max seq ize and max seqs per poa as mandatory in python API. and accept the rest as option keyword args with default params. if default params are provided, then we use the 2 argument constructor for batch size, otherwise we use the 5 arg version. we expose all params, but make some optional and some mandatory. only thing is if some of the optional ones are provided, all the others need to be as well.

Reply:
@rilango - I personally don't like to expose structs as APIs since it's hard to track API changes as structs themselves change. in python the keyword args allows us to add new options without changing the mandatory args

Reply:
I see, I agree with your comment regarding exposing structs in API.

Reply:
Actually that was my point :) Avoid API method prototype if the underlying structure changes. But I get the point, python's keyword and default values this will also make batchsize completely optional. I will update a new version in a moment..